airflow/models/taskinstance.py

```diff
@@ -1103,7 +1108,7 @@ def _handle_reschedule(self, actual_start_date, reschedule_exception, test_mode=
         self.log.info('Rescheduling task, marking task as UP_FOR_RESCHEDULE')
 
     @provide_session
-    def handle_failure(self, error, test_mode=None, context=None, session=None):
+    def handle_failure(self, error, test_mode=None, context=None, session=None, force_fail=False):
```

Comment:
A style point: please keep session as the final argument.

Reply:
10-4, will change!